File: DemoMethods/PythonDemo/demo_methods/Basic/Query1.py
from demo.holder import DocumentStoreHolder
from demo.entities import *


class Query1(object):
    @staticmethod
    def query1(country="UK"):
        with DocumentStoreHolder.get_store().open_session() as session:
            query, stats = list(session.query(object_type=Company, with_statistics=True,
                                              nested_object_types={"Address": AddressC, "Contact": ContactC}).where(
                **{"Address.Country": country}))

            return query[0] if query else None

File: DemoServer/Demos/Basic/Query1.cs
using System;
using System.Linq;
using DemoServer.Controllers;
using DemoServer.Entities;
using DemoServer.Helpers;
using Microsoft.AspNetCore.Mvc;
using Raven.Client.Documents.Session;

namespace DemoServer.Demos.Basic
{
    public partial class BasicController : BaseController
    {
        [HttpGet]
        [Route("/basic/query1")]
        [Demo("Query I", DemoOutputType.Flatten, demoOrder: 40)]
        public object Query1(string country = "UK")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                QueryStatistics stats;

                var query =
                    from company in session.Query<Company>().Statistics(out stats)
                    where company.Address.Country == country
                    select company;

                var result = query.FirstOrDefault();

                ServerTime = TimeSpan.FromMilliseconds(stats.DurationMilliseconds);

                return result;
            }
        }
    }
}

